src/bank/two/DebtAccountDecorator.java

```diff
@@ -0,0 +1,76 @@
+package bank.two;
+
+public class DebtAccountDecorator extends AccountDecorator {
+
+    private int debtLimit;
+    private int debtBalance;
+
+    public DebtAccountDecorator(Account account, int debtLimit) {
+        super(account);
+        this.debtLimit = debtLimit;
+        this.debtBalance = 0;
+    }
+
+    private void executeCommand(Command newCommand) {
+        newCommand.execute();
+    }
+
+    @Override
+    public int getBalance() {
+        if(debtBalance > 0) return (-1) * debtBalance;
+        else return account.getBalance();
+    }
+    public void depositMoney(int amount) {
+        executeCommand(new Deposit(this, amount));
+    }
+    @Override
+    public void withdrawMoney(int amount) {
+        executeCommand(new Withdrawal(this,amount));
+    }
+    @Override
+    public void transferMoney(Account dstAccount,int amount) {
+        Transfer transfer = new Transfer(this, dstAccount, amount);
+        executeCommand(transfer);
+        if(transfer.checkStatus()) {
+            account.getBank().addTransferToSession(transfer);
+        }
+    }
+    @Override
+    public void increaseBalance(int amount){
+        if(debtBalance > 0 && amount < debtBalance) {
+            debtBalance -= amount;
+        }
+        else if(debtBalance > 0 && amount > debtBalance) {
+            account.increaseBalance(amount-debtBalance);
+            debtBalance = 0;
```

Comment:
Taka konstrukcja to już trochę monster if, znacznie lepszym wyjściem jest wstawienie return; na końcu bloku if/else if, wtedy odchodzi nam else na samym końcu - bo albo metoda zakończy wykonywanie na bloku warunkowym albo w ogóle w niego nie wejdzie i wykona kod który jest poza nim. Nie jest to błąd, po prostu drobna uwaga co do estetyki kodu. :)